### ISSUE TITLE
Same result if a commit ended up being commit or rollback

### ISSUE THREAD

**Issue by mcherep** [epgsql/epgsql#112]
The execution of commit with `epgsql` and `epgsqla` always returns ok and it makes no difference if a rollback occurred due to errors in the transaction. An example:

```erlang
{ok, [], []} = epgsql:squery(C, "BEGIN;"),
%% The table does not exist and the error fails the transaction
{error, _} = epgsql:squery(C, "SELECT * FROM undefined_table;"),
%% A rollback is executed, but it is impossible to know
{ok, [], []} = epgsql:squery(C, "COMMIT;").
```

This issue is related to the pull request #111.

A working solution proposed by @seriyps is:

```erlang
Ref = epgsqli:squery(C,"COMMIT"),
CompleteCmd = receive {C, Ref, {complete, CompleteCommand}} -> CompleteCommand end,
receive {C, Ref, done} -> ok end,
case CompleteCmd of
    commit -> ok;
    rollback -> {error, rollback}
end.
```

**Comment by seriyps** [comment]
What I personally propose is to store `CommandComplete` status in epgsql's `#state{}` and add function to get it's value.

In that case someone, who is interested in transaction be sure committed, should do smth like this:

```
{ok, [], []} = epgsql:squery(C, "COMMIT"),
# not yet sure about name. In libpq this function called 'PQcmdStatus'
case epgsql:get_cmd_status(C) of
    {ok, commit} -> ok;
    {ok, rollback} -> {error, rollback}
end
```

Which is much more simple than dealing with this async stuff.

Also we can export `txstatus` state variable (`PQtransactionStatus` in libpq), but that's not really related.

**Comment by seriyps** [comment]
Relevant topic https://groups.google.com/forum/#!topic/epgsql/ePdg5-J6XZA

**Comment by mcherep** [comment]
I'd rather use the solution that you propose now instead of async calls. I'll take a look at this.

**Comment by seriyps** [comment]
Or you may wait a bit. I can implement this in 1-2 days

**Comment by seriyps** [comment]
Looks like it was resolved by #114
